signing-and-verification-workflow.md

```diff
@@ -16,7 +16,7 @@ The user wants to sign an OCI artifact and push the signature to a repository.
     1. Verify that the signing certificate is valid and satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
     1. Verify that the signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
     1. Generate signature.
-        1. Generate signature using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes).
+        1. Generate signature using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes). Also, as part of this step, the user-defined/supplied custom attributes should be added to the signature's signed attributes.
```

Comment:
The term `signature's signed attributes` is ambiguous. Current solution for user-defined attributes is to store them in the `annotations` field of the content descriptor where the descriptor is the signing payload. Meanwhile, modern signature synax / envelopes like JWS, PKCS#7, COSE support protected headers besides the payload. The protected headers are the signature's signed attributes.  They are signed with the signing payload but not the part of the signing payload.

Reply:
We have defined signed attributes in [signature-specification.md#signature-envelope](https://github.com/notaryproject/notaryproject/blob/958e868747a78b889eef99f12f604b2e50de4aed/signature-specification.md#signature-envelope). The signed attribute matters in systems that support multiple signatures in a single signature envelope, in such cases every signature shares a common payload but have different signed attributes. As in the case of notary we support 1 signature per envelope it doesn't matters if it goes into payload or in dedicated signed attributes filed.